tests/callbacks/test_early_stopping.py

```diff
@@ -260,7 +260,7 @@ def validation_epoch_end(self, outputs):
         max_epochs=10,
     )
     trainer.fit(model)
-    assert trainer.current_epoch == expected_stop_epoch
+    assert trainer.current_epoch - 1 == expected_stop_epoch
```

Comment:
So this test has `expected_stop_epoch=2` which is used to index a list. Thus we know that this is a 0-based index. 2 idx == 3 count.

This means that 3 epochs will run and then training will stop.

The behaviour doesn't change between master and this PR. What changed is that we `completed++` now in `FitLoop.on_run_end` because this removed `        self.current_epoch = max(self.current_epoch - 1, 0)`

Reply:
`expected_stop_epoch` is confusing because it should be named `expected_stop_epoch_index` by the way it's used.